Doc for vSphere provider (#3000)
skypilot-org/skypilot: docs/source/cloud-setup/cloud-permissions/vsphere.rst

    :align: center
    :alt: Storage Policy Navigate New

6. In the **Name and description** step, specify the name as ``skypilot_policy``.

.. image:: ../../images/screenshots/vsphere/vsphere-vm-storage-policy-name.png
    :width: 80%
    :align: center
    :alt: Policy Name

7. In the **Policy structure** step, select **Enable tag based placement rules**.

.. image:: ../../images/screenshots/vsphere/vsphere-vm-storage-policy-rule.png
    :width: 80%
    :align: center
    :alt: Policy Rule

8. In the **Tag based placement** step, select the tag **skypilot**.

.. image:: ../../images/screenshots/vsphere/vsphere-vm-storage-policy-tag.png
    :width: 80%
    :align: center
    :alt: Policy Tags

9. In the **Storage compatibility** step, review the datastores.

.. image:: ../../images/screenshots/vsphere/vsphere-vm-storage-policy-review.png
    :width: 80%
    :align: center
    :alt: Policy Review

10. Review and Click the **FINISH** to create the policy.

.. _cloud-prepare-vsphere-content-library-create:

Prepare VM image
~~~~~~~~~~~~~~~~

The VM must be Linux-based; we take Ubuntu 20.04 as the base OS in this document. Other Linux distributions may also works but not promised.

> **Step1. Prepare a linux-based VM**.

Reviewer: ```suggestion
**Step 1. Prepare a linux-based VM**.
```

Open web browser and Login your vSphere Client, create a linux-based Virtual Machine.

If you're unfamiliar with the process of creating a virtual machine, please refer to this guide for assistance: `Deploying Virtual Machines <https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vm_admin.doc/GUID-39D19B2B-A11C-42AE-AC80-DDA8682AB42C.html>`_

> **Step2. Create a user and Enable password-less sudo**.

Reviewer: ```suggestion
**Step 2. Create a user and Enable password-less sudo**.
```

Login your prepared VM, Create a user named ``ubuntu``, and add to the sudo group.

.. code-block:: shell

    sudo adduser ubuntu
    sudo usermod -aG sudo ubuntu

Edit the sudoer file to enable passwordless sudo.

.. code-block:: shell

    sudo visudo

Add the following line:

.. code-block:: text

    ubuntu   ALL=(ALL:ALL) NOPASSWD: ALL

> **Step3. Check VMtools version**.

Reviewer: ```suggestion
**Step 3. Check VMtools version**.
```

Execute the following command to check VMtools version 10.1.0 or later is required.

.. code-block:: shell

    vmware-toolbox-cmd -v

> **Step4. Enabling Custom Scripts in VMtools**.

Reviewer: ```suggestion
**Step 4. Enabling Custom Scripts in VMtools**.
```

Execute the following command to check if the enable-custom-scripts option is enabled:

.. code-block:: shell

    sudo vmware-toolbox-cmd config get deployPkg enable-custom-scripts

If the enable-custom-scripts option is disabled, enable it by executing:

.. code-block:: shell

    sudo vmware-toolbox-cmd config set deployPkg enable-custom-scripts true

> **Step5. Install the GPU driver**.

Reviewer: ```suggestion
**Step 5. Install the GPU driver**.
```

Different GPUs require different drivers; please choose the correct driver based on your GPU mode and OS version.
Take Ubuntu and Nvidia GPU for example, the following driver works for most of the case:

.. code-block:: shell

    sudo apt install nvidia-headless-535-server --no-install-recommends

> **Step6. Convert the VM to template**.

Reviewer: ```suggestion
**Step 6. Convert the VM to template**.
```

1. Power off the VM.

2. Select the VM and right click -> **Clone**-> **Clone as Template to Library**.

.. image:: ../../images/screenshots/vsphere/vm-clone-to-template.png
    :width: 80%
    :align: center
    :alt: VM Clone To Temple

3. Select Template type: **OVF**, provide the template name and click **Next**.

.. image:: ../../images/screenshots/vsphere/vm-clone-to-template-ovf.png
    :width: 80%
    :align: center
    :alt: VM Clone To Temple OVF

4. Select a Content Library location and click **Next**.

   Create one if you haven't set up a **Local** content library. Please refer to `Create a Library <https://docs.vmware.com/en/VMware-vSphere/7.0/com.vmware.vsphere.vm_admin.doc/GUID-2A0F1C13-7336-45CE-B211-610D39A6E1F4.html>`_

.. image:: ../../images/screenshots/vsphere/vm-clone-to-template-cl.png
    :width: 80%
    :align: center
    :alt: VM Clone To Temple Content Lib

5. Review and click **Finish**.

> **Step7. Tag the VM template**.

Reviewer: ```suggestion
**Step 7. Tag the VM template**.
```
